src/BinSkim.Driver/RoslynAnalyzer/ILImporter.cs

```diff
@@ -0,0 +1,741 @@
+        }
+
+        private void ImportNop()
+        {
+        }
+
+        private void ImportBreak()
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportLoadVar(int index, bool argument)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportStoreVar(int index, bool argument)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportAddressOfVar(int index, bool argument)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportDup()
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportPop()
+        {
+            Append(new ExpressionStatement(Pop().Expression));
+        }
+
+        private void ImportJmp(int token)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportCasting(ILOpcode opcode, int token)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportCall(ILOpcode opcode, int token)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportLdFtn(int token, ILOpcode opCode)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportLoadInt(long value, StackValueKind kind)
+        {
+            object boxedValue;
+            ITypeSymbol type;
+
+            if (kind == StackValueKind.Int32)
+            {
+                boxedValue = (int)value;
+                type = Int32Type;
+            }
+            else
+            {
+                Debug.Assert(kind == StackValueKind.Int64);
+                boxedValue = value;
+                type = Int64Type;
+            }
+
+            Push(kind, new LiteralExpression(boxedValue, type));
+        }
+
+        private void ImportLoadFloat(double value)
+        {
+            Push(StackValueKind.Float, new LiteralExpression(value, DoubleType));
+        }
+
+        private void ImportLoadNull()
+        {
+            Push(StackValueKind.ObjRef, new LiteralExpression(null, ObjectType));
+        }
+
+        private void ImportReturn()
+        {
+            Append(new ReturnStatement(_method.ReturnsVoid ? null : Pop().Expression));
+        }
+
+        private void ImportFallthrough(BasicBlock next)
+        {
+            ReqireEmptyStackForNow();
+            MarkBasicBlock(next);
+        }
+
+        private void ImportSwitchJump(int jmpBase, int[] jmpDelta, BasicBlock fallthrough)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportBranch(ILOpcode opcode, BasicBlock target, BasicBlock fallthrough)
+        {
+            StackValue left, right;
+
+            if (opcode == ILOpcode.brtrue || opcode == ILOpcode.brfalse)
+            {
+                left = Pop();
```

Comment:
ok, I get this code. I think I would have authored this to conditionally create a one or a zero literal based on input opcode, and then change opcode for both cases to ILOpcode.beq. :P